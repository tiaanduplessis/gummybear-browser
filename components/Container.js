import React from 'react'
import PropTypes from 'prop-types'
import { View, StyleSheet } from 'react-native'

import colors from '../constants/colors'

const Container = ({
  style,
  children,
  padding,
  backgroundColor,
  ...otherProps
}) => {
  const containerStyles = { padding, backgroundColor }

  return (
    <View style={[styles.container, containerStyles]} {...otherProps}>
      {children}
    </View>
  )
}

Container.propTypes = {
  padding: PropTypes.number,
  backgroundColor: PropTypes.string,
  center: PropTypes.bool
}

Container.defaultProps = {
  padding: 0,
  backgroundColor: colors.white
}

const styles = StyleSheet.create({
  container: { flex: 1 }
})

export default Container
