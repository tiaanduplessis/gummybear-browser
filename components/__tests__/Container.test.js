import 'react-native'
import React from 'react'
import Container from '../Container'
import renderer from 'react-test-renderer'

it('renders correctly', () => {
    const tree = renderer.create(<Container />).toJSON()
    expect(tree).toMatchSnapshot()
})
