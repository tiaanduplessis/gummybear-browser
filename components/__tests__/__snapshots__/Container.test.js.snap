// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Array [
      Object {
        "flex": 1,
      },
      Object {
        "backgroundColor": "#fff",
        "padding": 0,
      },
    ]
  }
/>
`;
